packages/js-sdk/src/sandbox/index.ts

```diff
@@ -344,26 +367,77 @@ export class Sandbox extends SandboxApi {
    *
    * @param path the directory where to upload the file, defaults to user's home directory.
    *
+   * @param useSignature URL will be signed with the access token, this can be used for uploading files to the sandbox from a different environment (e.g. browser).
+   *
+   * @param signatureExpirationInSeconds URL will be signed with the access token, this can be used for uploading files to the sandbox from a different environment (e.g. browser).
+   *
    * @returns URL for uploading file.
    */
-  uploadUrl(path?: string) {
-    return this.fileUrl(path)
+  uploadUrl(path?: string, useSignature?: boolean, signatureExpirationInSeconds?: number) {
+    if (!this.envdAccessToken && (useSignature != undefined || signatureExpirationInSeconds != undefined)) {
```

Comment:
what if `useSignature` is `false`. It's little bit weird, but you could have it as (env) variable 

Convert to boolean please